File: Gemini/3. Patrón de Diseño Singleton/Ruby/singleton.rb
require 'singleton'

class MiSingleton
  include Singleton
  attr_accessor :valor

  def initialize(valor=nil)
    @valor = valor
  end
end

# Prueba
instancia1 = MiSingleton.instance
instancia1.valor = "Primer valor"
puts "Valor de instancia1: #{instancia1.valor}"

instancia2 = MiSingleton.instance
instancia2.valor = "Segundo valor"
puts "Valor de instancia2: #{instancia2.valor}"
puts "¿Instancia1 es la misma que instancia2? #{instancia1.equal?(instancia2)}"
puts "Valor de instancia1 después de modificar instancia2: #{instancia1.valor}"

File: Gemini/3. Patrón de Diseño Singleton/Python/singleton.py
class Singleton:
    _instance = None

    def __new__(cls, *args, **kwargs):
        if not cls._instance:
            cls._instance = super(Singleton, cls).__new__(cls, *args, **kwargs)
            # Inicialización de la instancia (opcional)
            cls._instance.valor = None
        return cls._instance

    def asignar_valor(self, valor):
        self.valor = valor

    def obtener_valor(self):
        return self.valor

# Prueba
instancia1 = Singleton()
instancia1.asignar_valor("Primer valor")
print(f"Valor de instancia1: {instancia1.obtener_valor()}")

instancia2 = Singleton()
instancia2.asignar_valor("Segundo valor")
print(f"Valor de instancia2: {instancia2.obtener_valor()}")
print(f"¿Instancia1 es la misma que instancia2? {instancia1 is instancia2}")
print(f"Valor de instancia1 después de modificar instancia2: {instancia1.obtener_valor()}")